src/components/API.js

```diff
@@ -0,0 +1,73 @@
+export default class API {
+  constructor(options) {
+    this.server = options.baseUrl;
+    this.headers = options.headers;
+  }
+
+  _checkRes(res) {
+    if (res.ok) {
+      return res.json();
+    }
+    return Promise.reject(`Error ${res.status}`);
+  }
+
+  getInitialCards() {
+    return fetch(`${this.server}/cards`, {
+      headers: this.headers,
+    }).then(this._checkRes);
```

Comment:
**Could be improved**

You can make a special method (function ) for fetching and checking responses **not to duplicate** it in every request:

```js

  _request(url, options) {
    return fetch(url, options).then(this._checkResponse)
  }

```

And now you can safely replace `fetch` with `this._request` and remove repeated checking. And you don’t need to change anything else.